### PR TITLE
Issue #3258883: Revoke select account cancellation method permission for authenticate…

### PR DESCRIPTION
**Pull request by NickDaelemans**
…d users.

## Problem
If I disable my account as an authenticated user I have to choose a cancellation method when cancelling my own account.

## Solution
We should skip this step by removing the permissions for authenticated users. The desired cancellation method can be selected from within the user settings.

## Issue tracker
https://www.drupal.org/project/social/issues/3258883
## How to test
- [ ] Log in as an authenticated user
- [ ] Cancel your account (Settings -> Cancel account)
- [ ] You should not be able to select a cancellation method.

## Release notes
By revoking the `select account cancellation method` authenticated users don't have to select a cancellation method anymore when cancelling their own account.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2693-ejsxggxanhxqisqkigfkbywbdzr0tuvb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61e537f8f8a118577f2a4ebb

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2693-ejsxggxanhxqisqkigfkbywbdzr0tuvb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61e537f8f8a118577f2a4ebb

File: modules/social_features/social_user/social_user.install [hunk 1]
```diff
@@ -228,6 +226,15 @@ function social_user_update_8810() {
...
 }
 
+/**
+ * Revoke the permission select account cancellation method.
+ */
+function social_user_update_8811(): void {
+  user_role_revoke_permissions(RoleInterface::AUTHENTICATED_ID, [
+    'select account cancellation method',
+  ]);
+}
```
Comment (nkoporec) [inline modules/social_features/social_user/social_user.install:236]: Why duplicated update hook? If the intention here is to be backward compatible, then you should create a separate PR to 11.0.x and 10.3.x with the correct hook number.

**Review by nkoporec — CHANGES_REQUESTED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2693-ejsxggxanhxqisqkigfkbywbdzr0tuvb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61e537f8f8a118577f2a4ebb

**Review by nkoporec — APPROVED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2693-ejsxggxanhxqisqkigfkbywbdzr0tuvb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61e537f8f8a118577f2a4ebb

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2693-ejsxggxanhxqisqkigfkbywbdzr0tuvb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61e537f8f8a118577f2a4ebb

**Pull request merged**